src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -4863,6 +4863,29 @@ public static JavaScriptException throwCustomError(
         return new JavaScriptException(error, filename, linep[0]);
     }
 
+    /**
+     * Implement the ECMAScript abstract operation "SpeciesConstructor" defined in section 7.2.33 of
```

Comment:
Consider moving this to o.m.j.AbstractEcmaObjectOperations, as it is one of the Abstract Object Operations defined by EcmaScript. When you do, please add a first param of type Context (as per the instructions outlines at the top of o.m.j.AbstractEcmaObjectOperations): while the function might not need Context atm, it might in the future

Consider adding a link to the specific section in the specific version of the EcmaScript spec implemented: the codebase is full of textual references to a section in the spec, but as the spec evolves, new chapters get added and things get renamed, so textual reference without context become meaningfull. A link to [ES2021 Species Constructor](https://262.ecma-international.org/12.0/#sec-speciesconstructor) however will (hopefully) remain valid and allows someone looking at this code x years from now to figure out the delta between the specific version implemented and the latest spec at that point in time